File: leetcode/template/0901-1000/0965.univalued-binary-tree.js
/**
 * https://leetcode-cn.com/problems/univalued-binary-tree/
 * 0965.单值二叉树
 * 难度：容易
 * 
 * 如果二叉树每个节点都具有相同的值，那么该二叉树就是单值二叉树。
 * 只有给定的树是单值二叉树时，才返回 true；否则返回 false。
 * 
 * 示例 1：
 * [示意图](https://github.com/Tairraos/practice/blob/master/leetcode/figs/0965-1.png)
 * 输入：[1,1,1,1,1,null,1]
 * 输出：true
 * 
 * 示例 2：
 * [示意图](https://github.com/Tairraos/practice/blob/master/leetcode/figs/0965-2.png)
 * 输入：[2,2,2,5,2]
 * 输出：false
 * 
 * 提示：
 *   - 给定树的节点数范围是 [1, 100]。
 *   - 每个节点的值都是整数，范围为 [0, 99] 。
 */

/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @return {boolean}
 */
var isUnivalTree = function(root) {
    
};

// Local test
let assert = require("assert");
console.time("leetcode");

assert.deepEqual(isUnivalTree("参数"), "期望结果");
assert.deepEqual(isUnivalTree("参数"), "期望结果");

console.log("哈哈哈哈，所有Case都通过了");
console.timeEnd("leetcode");

File: leetcode/template/0901-1000/0965.univalued-binary-tree.py
# https://leetcode-cn.com/problems/univalued-binary-tree/
# 0965.单值二叉树
# 难度：容易
# 
# 如果二叉树每个节点都具有相同的值，那么该二叉树就是单值二叉树。
# 只有给定的树是单值二叉树时，才返回 true；否则返回 false。
# 
# 示例 1：
# [示意图](https://github.com/Tairraos/practice/blob/master/leetcode/figs/0965-1.png)
# 输入：[1,1,1,1,1,null,1]
# 输出：true
# 
# 示例 2：
# [示意图](https://github.com/Tairraos/practice/blob/master/leetcode/figs/0965-2.png)
# 输入：[2,2,2,5,2]
# 输出：false
# 
# 提示：
#   - 给定树的节点数范围是 [1, 100]。
#   - 每个节点的值都是整数，范围为 [0, 99] 。


# Definition for a binary tree node.
class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None

class Solution:
    def isUnivalTree(self, root: TreeNode) -> bool:
        "put solution here"


# Local test
if __name__ == '__main__':
    import time
    launch_start = time.time()
    unit = Solution()

    assert unit.isUnivalTree('param') == 'expect', 'Case 1'
    assert unit.isUnivalTree('param') == 'expect', 'Case 2'

    print('Good job! We have passed all test case.')
    print('Time cast: ' + str(int((time.time() - launch_start) * 1000000)/1000) + 'ms')
